core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4686,8 +4686,17 @@ private void rewriteOrderByPosition(final QueryModel model) throws SqlException
                 final ExpressionNode orderBy = orderByNodes.getQuick(i);
                 final CharSequence column = orderBy.token;
 
-                char first = column.charAt(0);
-                if (first < '0' || first > '9') {
+                // assess whether the expression is entirely numeric, or not
```

Comment:
we can do this "optimistically" instead of scanning column name twice. The previous check would suffice. The change could be to "continue" when catching number parse exception, it yields the same result with less overhead